src/Adapter/LinearRegressionAdapter.php

```diff
@@ -2,6 +2,102 @@
 
 namespace Zeeml\Algorithms\Adapter;
 
+use Zeeml\Algorithms\Dataset\DatasetInterface;
+
 class LinearRegressionAdapter extends AbstractAdapter
 {
+    protected $intercept;
+    protected $slope;
+
+    /**
+     * LinearRegressionAdapter constructor.
+     * @param DatasetInterface $dataset
+     */
+    public function __construct(array $data)
```

Comment:
a solution would be to provide both possibilities, user could pass an array or a dataset. if the argument is an array, we could process it as a dataset internally.
I think the overhead could be low but imagine the ability to pass the dataset object to another process to display it ($graph = new Graph($dataset));